utm/conversion.py

```diff
@@ -278,7 +281,9 @@ def from_latlon(latitude, longitude, force_zone_number=None, force_zone_letter=N
                                         a4 / 24 * (5 - lat_tan2 + 9 * c + 4 * c**2) +
                                         a6 / 720 * (61 - 58 * lat_tan2 + lat_tan4 + 600 * c - 330 * E_P2)))
 
-    if not northern:
+    if force_northern is None and mixed_signs(latitude):
```

Comment:
This should probably also handle the case of a set `zone_letter` to also ignore mixed_signs check.